### PR TITLE
feat(orchestrator): add column filters

### PR DESCRIPTION
**Pull request by gciavarrini**
The orchestrator UI needs to filter by any column.

At this time only `in, like, equal, isnull` operator for `string` are supported.
`FieldFilters` can be combined by `LogicalFilters` (`and`, `or`).
Filters are available for `/v2/workflows/instances` and `/v2/workflows/overview` endpoints.

#### Equal
```bash
curl --request POST \
  --url http://localhost:7007/api/orchestrator/v2/workflows/overview \
   -H "Authorization: $token" \
  --header 'Content-Type: application/json' \
  --data '{
  "filters": {
    "operator": "OR",
    "filters": [
      {
        "field": "name",
        "operator": "EQ",
        "value": "Hello World Workflow"
      },
      {
        "field": "id",
        "operator": "EQ",
        "value": "yamlgreet"
      }
    ]
  },
  "paginationInfo": {
    "offset": 0,
    "pageSize": 10
  }
}'
```

returns 

```json
{"items":[{"id":"9fa2a881-c932-468d-83a9-687b9f1e62a7","processId":"hello_world","processName":"Hello World Workflow","businessKey":null,"category":"infrastructure","start":"2024-10-03T10:23:32.654Z","end":"2024-10-03T10:23:32.645Z","duration":"a few seconds","workflowdata":{"myKey":"myValue","mantra":"Serverless Workflow is awesome!","greeting":"Hello World"},"status":"Completed","nodes":[{"id":"e298367d-cbfa-4af7-a454-b1a186cbe25e","__typename":"NodeInstance"},{"id":"36d5d322-b711-472f-9a3f-23f5a74d6904","__typename":"NodeInstance"},{"id":"1e7da447-c8c4-4918-9a31-539b1f1eba6b","__typename":"NodeInstance"},{"id":"22b8c488-7858-4c38-8d31-17cd500ed7eb","__typename":"NodeInstance"}]}],"paginationInfo":{"pageSize":10,"offset":0,"totalCount":1}}
```
#### Like

```bash
curl --request POST \
  --url http://localhost:7007/api/orchestrator/v2/workflows/instances \
  -H "Authorization: $token" \
  --header "Content-Type: application/json" \
  --data '{
    "filters": {
      "field": "processName",
      "operator": "LIKE",
      "value": "Hello%"
    },
    "paginationInfo": {
      "offset": 0,
      "pageSize": 10
    }
  }'
```
returns
```json
{"items":[{"id":"9fa2a881-c932-468d-83a9-687b9f1e62a7","processId":"hello_world","processName":"Hello World Workflow","businessKey":null,"category":"infrastructure","start":"2024-10-03T10:23:32.654Z","end":"2024-10-03T10:23:32.645Z","duration":"a few seconds","workflowdata":{"myKey":"myValue","mantra":"Serverless Workflow is awesome!","greeting":"Hello World"},"status":"Completed","nodes":[{"id":"e298367d-cbfa-4af7-a454-b1a186cbe25e","__typename":"NodeInstance"},{"id":"36d5d322-b711-472f-9a3f-23f5a74d6904","__typename":"NodeInstance"},{"id":"1e7da447-c8c4-4918-9a31-539b1f1eba6b","__typename":"NodeInstance"},{"id":"22b8c488-7858-4c38-8d31-17cd500ed7eb","__typename":"NodeInstance"}]}],"paginationInfo":{"pageSize":10,"offset":0,"totalCount":1}}
```
#### Equal
```bash
curl --request POST \
  --url http://localhost:7007/api/orchestrator/v2/workflows/instances \
  -H "Authorization: $token" \
  --header "Content-Type: application/json" \
  --data '{
    "filters": {
      "field": "processName",
      "operator": "EQ",
      "value": "Hello World Workflow"
    },
    "paginationInfo": {
      "offset": 0,
      "pageSize": 10
    }
  }'
```
returns
```json
{"items":[{"id":"9fa2a881-c932-468d-83a9-687b9f1e62a7","processId":"hello_world","processName":"Hello World Workflow","businessKey":null,"category":"infrastructure","start":"2024-10-03T10:23:32.654Z","end":"2024-10-03T10:23:32.645Z","duration":"a few seconds","workflowdata":{"myKey":"myValue","mantra":"Serverless Workflow is awesome!","greeting":"Hello World"},"status":"Completed","nodes":[{"id":"e298367d-cbfa-4af7-a454-b1a186cbe25e","__typename":"NodeInstance"},{"id":"36d5d322-b711-472f-9a3f-23f5a74d6904","__typename":"NodeInstance"},{"id":"1e7da447-c8c4-4918-9a31-539b1f1eba6b","__typename":"NodeInstance"},{"id":"22b8c488-7858-4c38-8d31-17cd500ed7eb","__typename":"NodeInstance"}]}],"paginationInfo":{"pageSize":10,"offset":0,"totalCount":1}}
```

## Jira issue
Fix [FLPATH-1757](https://issues.redhat.com/browse/FLPATH-1757)

### CODE REVIEW

**Comment by batzionb** [comment]
hi,
I tried this post request and got error

`curl --request POST \
  --url http://localhost:7007/api/orchestrator/v2/workflows/instances \
  -H "Authorization: Bearer $token" \
  --header "Content-Type: application/json" \
  --data '{
    "filters": {
      "field": "processName",
      "operator": "EQ",
      "value": "Wait or Error"
    },
    "paginationInfo": {
      "offset": 0,
      "pageSize": 10
    }
  }'
`

what am I missing?

**Comment by gciavarrini** [comment]
> hi, I tried this post request and got error

@batzionb there was a bug. Can you retry? I also added another example in the PR description.

**Comment by batzionb** [comment]
That post command works now

Now I tried the LIKE operator and I get an empty list

 ```
curl --request POST \
  --url http://localhost:7007/api/orchestrator/v2/workflows/instances \
  -H "Authorization: Bearer $token" \
  --header "Content-Type: application/json" \
  --data '{
    "filters": {
      "field": "processName",
      "operator": "LIKE",
      "value": "Wait"
    },
    "paginationInfo": {
      "offset": 0,
      "pageSize": 10
    }
  }'


```

File: plugins/orchestrator-backend/src/helpers/queryBuilder.ts
Comment (ydayagi) [inline plugins/orchestrator-backend/src/helpers/queryBuilder.ts:1]: what about simply letting the user put a graphql query in the request?

File: plugins/orchestrator-backend/src/helpers/queryBuilder.ts [hunk 1]
```diff
@@ -53,8 +59,121 @@ function buildPaginationClause(pagination?: Pagination): string {
...
+    return `${filters.field}: {${getGraphQLOperator(filters.operator)}: ${booleanValue}}`;
+  }
+
+  if (Array.isArray(value)) {
+    value = `[${value.map(v => formatValue(filters.field, v, introspection)).join(', ')}]`;
+  } else {
+    value = formatValue(filters.field, value, introspection);
+  }
+
+  if (
```
Comment (ydayagi) [inline plugins/orchestrator-backend/src/helpers/queryBuilder.ts:105]: the call to getGraphQLOperator already tests whether the given operator is supported

File: plugins/orchestrator-backend/src/helpers/queryBuilder.ts
Comment (gciavarrini) [reply plugins/orchestrator-backend/src/helpers/queryBuilder.ts:1]: We had a discussion about this, the idea is to let the client (our orchestrator fronted or any other client) unaware about the existence of GraphQL. It's an implementation detail. What if, for example, there is a change in the GraphQL schema? It shouldn't be a user problem. In addition, writing GraphQL query is very error-prone.

File: plugins/orchestrator-backend/src/helpers/queryBuilder.ts (hunk 1 above)
Comment (gciavarrini) [reply plugins/orchestrator-backend/src/helpers/queryBuilder.ts:105]: I don't understand this comment, need clarifications.

If you refer to this code block
```js
  if (
    filters.operator === FieldFilterOperatorEnum.Eq ||
    filters.operator === FieldFilterOperatorEnum.Like ||
    filters.operator === FieldFilterOperatorEnum.In
  ) {
    return `${filters.field}: {${getGraphQLOperator(filters.operator)}: ${value}}`;
  }
```
It's necessary to properly build the GraphQL condition, depending on the operator. 
In other words, `eq` `like` and `in` have different syntax then `isNull`.

**Comment by gciavarrini** [comment]
@batzionb @ydayagi PTAL

**Pull request merged**